source/m3_function.h

```diff
@@ -71,7 +71,7 @@ typedef struct M3Function
     u16                     numRetAndArgSlots;
 
     u16                     numLocals;                              // not including args
-    u16                     numLocalBytes;
+    u32                     numLocalBytes;
```

Comment:
There are many more `u16` fields here that, counting slots. A slot is 4 (or 8) bytes, so only increasing this variable isn't going to help much in the general case.

https://github.com/wasm3/wasm3/blob/35b5e2fb53c5cbc1ff3d7e42c381cd7cfa14f308/source/m3_compile.c#L2893

Reply:
This PR addresses the issue caused by this line. I don't think we need to change the slot count to `u32`, but I'm not sure which other variable types we might need to adjust. It makes more sense to me to add [`-fsanitize=integer`](https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html) in the CI.

Reply:
If you use 4 times as many locals as you are using now, you'll wrap around the `maxStackSlots` slot counter. So just changing this line only moves the problem by a factor of 4, not 2^16. Is that good enough?

> It makes more sense to me to add [-fsanitize=integer](https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html) in the CI.

Fixing the overflows (by returning error) and enabling UBSan does sound nice.

Reply:
> If you use 4 times as many locals as you are using now, you'll wrap around the `maxStackSlots` slot counter. So just changing this line only moves the problem by a factor of 4, not 2^16. Is that good enough?

I can verify that when I set `d_m3MaxFunctionStackHeight` to 32767, it first encounters `m3Err_functionStackOverflow` in `AllocateSlotsWithinRange`.

Reply:
Thanks for checking!